modules/core/src/v2/coins/xlm.ts

```diff
@@ -896,26 +936,44 @@ export class Xlm extends BaseCoin {
         operation.type === 'createAccount' || operation.type === 'payment'
       );
 
-      if (_.isEmpty(outputOperations)) {
-        throw new Error('transaction prebuild does not have any operations');
-      }
-
-      _.forEach(txParams.recipients, (expectedOutput, index) => {
-        const expectedOutputAddress = self.getAddressDetails(expectedOutput.address);
-        const output = outputOperations[index] as (stellar.Operation.Payment | stellar.Operation.CreateAccount);
-        if (output.destination !== expectedOutputAddress.address) {
-          throw new Error('transaction prebuild does not match expected recipient');
+      if (txParams.type === 'trustline') {
+        const trustlineOperations = _.filter(tx.operations, ['type', 'changeTrust']) as [stellar.Operation.ChangeTrust];
```

Comment:
Nit: Consider extracting to separate private helper function `verifyTrustlineTransaction` - this is getting a bit dense.

Reply:
Incorrect type: Return type should be `stellar.Operation.ChangeTrust[]`, which is not equivalent to `[stellar.Operation.ChangeTrust]`.

The [] around the type means this is a tuple array, which is an array with a length which is known at compile time. I don't think that's the case here, since we check the length dynamically below.

Reply:
Changed both